### PR TITLE
#14590: Move sfpi off LFS

### PR DESCRIPTION
**Pull request by nathan-TT**
Remove tt_metal/third_party/sfpi submodule.
Add smarts to tt_metal/hw to download sfpi release into runtome/sfpi

### Ticket
https://github.com/tenstorrent/tt-metal/issues/14590

### Problem description
LFS bad, don't use it.

### What's changed
New reorg and release mechanism for sfpi -- see 
https://github.com/tenstorrent/sfpi/pull/24

Remove tt_metal/third_party/sfpi submodule
Add smarts to tt_metal/hw to lazily fetch a specific sfpi release (this patch names v0.0.0-lfsectomy, once we're set I'll create v12.0.0 -- numbering them after the GCC major/minor version numbering and use the micro number for our own updates.

### Checklist
- [YES] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hw/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,15 @@ set(TYPES
     kernel
 )
 
+add_custom_target(
```
[the remark this answers is not in the document]
Comment (tt-rkim) [reply tt_metal/hw/CMakeLists.txt:21]: Edited the above to use the more idiomatic `FetchContent_Declare` and `FetchContent_MakeAvailable`

This also requires us to not depend on `sfpi` as a target, as it's just now just a name in this file that `FetchContent` will always ensure is available before going ahead with building.

Running an example change here: https://github.com/tenstorrent/tt-metal/actions/runs/11749862356

[the remark this answers is not in the document]
Comment (blozano-tt) [reply tt_metal/hw/CMakeLists.txt:21]: I like this @tt-rkim. If CMake gives us fetching macro for free, we should use it, then we don't have to maintain this new script.

[the remark this answers is not in the document]
Comment (nathan-TT) [reply tt_metal/hw/CMakeLists.txt]: yoinked @tt-rkim 's code. Updated instructions in the sfpi repo (https://github.com/tenstorrent/sfpi/pull/24)

**Review by tt-rkim — APPROVED** [review]

**Comment by nathan-TT** [comment]
Decided not to number after GCC releases, just use an incrementing major number.  Started at 5 to avoid any meaning to it.

**Review by arikTT — APPROVED** [review]

**Review by tt-dma — APPROVED** [review]

**Comment by tt-rkim** [comment]
@warthog9 @olofj 
BTW - thought you would be delighted to see this has been merged in.

**Pull request merged**